keycloak/event-listeners/src/main/java/org/eclipse/sw360/keycloak/event/listener/service/Sw360UserService.java

```diff
@@ -10,109 +10,372 @@
+                return envValue;
+            }
+        }
+        return fallbackValue;
+    }
+    
+    private Properties loadProperties() {
+        Properties props = new Properties();
+        
+        // Try external file first
+        java.io.File externalFile = new java.io.File(SYSTEM_CONFIGURATION_PATH, PROPERTIES_FILE_PATH);
+        if (externalFile.exists()) {
+            try (java.io.FileInputStream input = new java.io.FileInputStream(externalFile)) {
+                props.load(input);
+                logger.debug("Loaded CouchDB properties from external file: {}", externalFile.getAbsolutePath());
+                return props;
+            } catch (IOException e) {
+                logger.warn("Error loading external CouchDB properties file: {}", e.getMessage());
+            }
+        }
+        
+        // Fallback to classpath resource
+        try (InputStream input = getClass().getResourceAsStream("/"+PROPERTIES_FILE_PATH)) {
+            if (input != null) {
+                props.load(input);
+                logger.debug("Loaded CouchDB properties from classpath: {}", PROPERTIES_FILE_PATH);
+            } else {
+                logger.warn("CouchDB properties file not found in classpath: {}", PROPERTIES_FILE_PATH);
+            }
+        } catch (IOException e) {
+            logger.error("Error loading CouchDB properties from classpath: {}", e.getMessage(), e);
+        }
+        return props;
+    }
+
+    /**
+     * Retrieves all users from the SW360 database.
+     * 
+     * @return List of all users, empty list if none found or on error
+     */
     public List<User> getAllUsers() {
         try {
-            UserService.Iface sw360UserClient = getThriftUserClient();
-            return sw360UserClient.getAllUsers();
+            logger.debug("Event listener retrieving all users from SW360 database");
+            List<User> users = connector.getAll(User.class);
+            logger.info("Event listener retrieved {} users from SW360 database", users.size());
+            return users;
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            logger.error("Event listener error retrieving all users from SW360 database", e);
+            return Collections.emptyList();
         }
     }
 
+    /**
+     * Retrieves a user by email address.
+     * 
+     * @param email the email address to search for
+     * @return User if found, null otherwise
+     */
     public User getUserByEmail(String email) {
+        if (email == null || email.trim().isEmpty()) {
+            logger.warn("Event listener attempted to get user with null or empty email");
+            return null;
+        }
+        
         try {
-            UserService.Iface sw360UserClient = getThriftUserClient();
-            return sw360UserClient.getByEmail(email);
+            logger.debug("Event listener retrieving user by email: {}", email);
+            User user = connector.get(User.class, email);
```

Comment:
Will this work as `email` is not used as user's id?

Should use something similar to `getByEmail()` function which queries the view `byEmail`.

https://github.com/eclipse-sw360/sw360/blob/558851af121fc0a423deeca4c4b0822e3f7a0af1/backend/common/src/main/java/org/eclipse/sw360/datahandler/db/UserRepository.java#L171-L174